tests/lib/rules/max-classes-per-file.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * @fileoverview Tests for max-classes-per-file rule.
+ * @author James Garbutt <https://github.com/43081j>
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/max-classes-per-file"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+
+ruleTester.run("max-classes-per-file", rule, {
+    valid: [
+        {
+            code: "class Foo {}",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var x = class {};",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "var x = 5;",
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "class Foo {}\nvar cls = class {};",
+            options: [1, { excludeClassExpressions: true }],
+            parserOptions: { ecmaVersion: 6 }
+        },
+        {
+            code: "class Foo {}\nclass Bar {}",
+            options: [2],
+            parserOptions: { ecmaVersion: 6 }
+        }
+    ],
+
+    invalid: [
+        {
+            code: "class Foo {}\nclass Bar {}",
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "maximumExceeded", type: "Program" }]
+        },
+        {
+            code: "var x = class {};\nvar y = class {};",
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "maximumExceeded", type: "Program" }]
+        },
+        {
+            code: "class Foo {}\nvar x = class {};",
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "maximumExceeded", type: "Program" }]
+        },
+        {
+            code: "class Foo {} class Bar {} class Baz {}",
+            options: [2],
+            parserOptions: { ecmaVersion: 6 },
+            errors: [{ messageId: "maximumExceeded", type: "Program" }]
+        }
```

Comment:
It would be good to have an invalid test where the number of `ClassDeclaration` nodes is too high, but there are also `ClassExpression` nodes that aren't counted. (e.g., option is set to 2, and there are three `ClassDeclaration` and one `ClassExpression`)